### PR TITLE
Removes the Menus configuration functionality

### LINKED ISSUE #16413: Remove the Menus configuration from Site Details Screen
### Desscription
iOS has support to configure Menus via a native interface that has come up as a point of confusion in recent User studies. We'd like to remove this menu (to match Android) and then re-introduce it in a more holistic way during FSE.

### PR DESCRIPTION
**Pull request by antonis**
Fixes #16413

Related PRs:
* `WordPressKit-iOS`: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/398
* `WordPress-iOS-Shared`: https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/293

### Description
Removes the Menus configuration functionality. This includes removing:
* the menu item from *My Sites* screen
* everything under `WordPress/Classes/ViewRelated/Menus/`
* `MenusService.h/m` and `MenusServiceTests.m`

To test:
1. Open the app and select the *My Site* screen
2. Scroll down to the *PERSONALIZE* section
3. **Verify** that only the *Themes* option is available

|Before|After|
|---|---|
|IMG_B28718BCB92A-1|IMG_BC6DFED6449B-1|

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
4. N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by chipsnyder — COMMENTED** [review]
Hey @antonis,
I was surprised this PR removed `WordPress/Resources/en.lproj/Localizable.strings` because that contains all of the English localized strings for the project. I'm assuming that was in error?

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m [hunk 1]
```diff
@@ -956,9 +956,6 @@ + (TracksEventPair *)eventPairForStat:(WPAnalyticsStat)stat
         case WPAnalyticsStatMediaServiceUploadCanceled:
             eventName = @"media_service_upload_canceled";
             break;
-        case WPAnalyticsStatMenusAccessed:
```
Comment (chipsnyder) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m:959]: I didn't have a chance to do a complete review today but leaving a note here that we might be able to clean up `WPAnalyticsStatMenusCreatedItem`, `WPAnalyticsStatMenusCreatedMenu`, `WPAnalyticsStatMenusDeletedMenu`, `WPAnalyticsStatMenusDeletedItem`, `WPAnalyticsStatMenusDiscardedChanges`, `WPAnalyticsStatMenusEditedItem`, `WPAnalyticsStatMenusOpenedItemEditor`, and `WPAnalyticsStatMenusSavedMenu`

**Comment by antonis** [comment]
Hey @chipsnyder 👋 

> I was surprised this PR removed `WordPress/Resources/en.lproj/Localizable.strings` because that contains all of the English localized strings for the project. I'm assuming that was in error?

You are right. I reverted editing the `Localizable.strings` with ca71a0e36d6d699dc88428f34396021e107eee0e

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m (hunk 1 above)
Comment (antonis) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m:959]: Good catch. Removed with 33ef2aa3727b34220c854f4f2ed77b06405a310a

**Comment by antonis** [comment]
This feature request is dropped as discussed in `pbArwn-2c8-p2#comment-3138`

**Pull request closed without merging**